LocalPackages/AIChat/Sources/AIChat/Public API/AIChatViewController.swift

```diff
@@ -0,0 +1,204 @@
+        imageView.translatesAutoresizingMaskIntoConstraints = false
+
+        let imageSize: CGFloat = 28
+        NSLayoutConstraint.activate([
+            imageView.widthAnchor.constraint(equalToConstant: imageSize),
+            imageView.heightAnchor.constraint(equalToConstant: imageSize)
+        ])
+
+        let titleLabel = UILabel()
+        titleLabel.text = UserText.aiChatTitle
+        titleLabel.font = UIFont.systemFont(ofSize: 17, weight: .semibold)
+        titleLabel.textColor = .white
+        let stackView = UIStackView(arrangedSubviews: [imageView, titleLabel])
+        stackView.axis = .horizontal
+        stackView.spacing = 8
+        stackView.alignment = .center
+        stackView.distribution = .fill
+
+        navigationItem.leftBarButtonItem = UIBarButtonItem(customView: stackView)
+
+        let closeButton = UIBarButtonItem(
+            image: UIImage(named: "Close-24"),
+            style: .plain,
+            target: self,
+            action: #selector(closeAIChat)
+        )
+        closeButton.tintColor = .white
+
+        navigationItem.rightBarButtonItem = closeButton
+    }
+
+
+    private func addWebViewController() {
+        guard webViewController == nil else { return }
+
+        let viewController = AIChatWebViewController(chatModel: chatModel)
+        viewController.delegate = self
+        webViewController = viewController
+
+        addChild(viewController)
+        view.addSubview(viewController.view)
+        viewController.view.translatesAutoresizingMaskIntoConstraints = false
+
+        NSLayoutConstraint.activate([
+            viewController.view.topAnchor.constraint(equalTo: view.safeAreaLayoutGuide.topAnchor),
+            viewController.view.bottomAnchor.constraint(equalTo: view.bottomAnchor),
+            viewController.view.leadingAnchor.constraint(equalTo: view.leadingAnchor),
+            viewController.view.trailingAnchor.constraint(equalTo: view.trailingAnchor)
+        ])
+
+        viewController.view.backgroundColor = .black
+        viewController.view.layer.cornerRadius = 20
+        viewController.view.layer.maskedCorners = [.layerMinXMinYCorner, .layerMaxXMinYCorner]
+        viewController.view.clipsToBounds = true
+
+        viewController.didMove(toParent: self)
+    }
+
+    private func removeWebViewController() {
+        webViewController?.removeFromParent()
+        webViewController?.view.removeFromSuperview()
+        webViewController = nil
+    }
+}
+
+// MARK: - Event handling
+extension AIChatViewController {
+
+    private func subscribeToCleanupPublisher() {
+        cleanupCancellable = chatModel.cleanupPublisher
+            .receive(on: DispatchQueue.main)
+            .sink { [weak self] in
+                self?.webViewController?.reload()
+                self?.timerPixelHandler.markCleanup()
+            }
+    }
+
+    @objc private func closeAIChat() {
+        chatModel.startCleanupTimer()
+        dismiss(animated: true)
+    }
```

Comment:
One note that is probably not a problem: when this VC is dismissed, it triggers `viewDidAppear` in `MainViewController`. This causes various things to happen, like `m.debug.app-did-show-ui-time-more` firing - I'd want to check with @jaceklyp whether this pollutes the data coming from that pixel, or whether this is not an issue.